lumicks/pylake/detail/image.py

```diff
@@ -90,29 +90,39 @@ def round_up(size, n):
         return pixels.reshape(-1, lines_per_frame, pixels_per_line).squeeze()
 
 
-def save_tiff(image, filename, dtype, clip=False):
-        """Save an RGB `image` to TIFF
-
-        This is a thin wrapper around `tifffile` with additional safety checks
-
-        Parameters
-        ----------
-        image : np.array
-            Image data. An RGB image is expected to have `shape == (w, h, 3)`.
-        filename : str
-            The name of the TIFF file where the image will be saved.
-        dtype : np.dtype
-            The data type of a single color channel in the resulting image.
-        clip : bool
-            If enabled, the photon count data will be clipped to fit into the desired `dtype`.
-            This option is disabled by default: an error will be raise if the data does not fit.
-        """
-        import tifffile
-
-        info = np.finfo(dtype) if np.dtype(dtype).kind == "f" else np.iinfo(dtype)
-        if not clip and (np.min(image) < info.min or np.max(image) > info.max):
-            raise RuntimeError(f"Can't safely export image with `dtype={dtype.__name__}` channels."
-                               f" Switch to a larger `dtype` in order to safely store everything"
-                               f" or pass `force=True` to clip the data.")
-
-        tifffile.imsave(filename, image.astype(dtype))
+def save_tiff(image, filename, dtype, clip=False, pixel_size=1.0, pixel_time=1.0):
```

Comment:
I would suggest making a dedicated class for metadata which can be passed to this function so that we can more easily extend it with more metadata without needing to add a lot of function arguments here. E.g. `pixel_size` does not necessarily need to be the same for x and y so that's already an extra argument. And we'll definitely be adding more in the future.

You can make that metadata class have properties which return the arguments expected by `tifffile.imsave()`.

Reply:
Makes sense. I've moved it into its own class, with a from_data that takes the json containing the metadata. Also added a test for it.